src/dbus/mod.rs

```diff
@@ -29,28 +34,22 @@ impl DBusService {
     }
 
     #[context("failed to start object server")]
-    fn start_object_server(&mut self) -> Result<()> {
-        let connection = zbus::Connection::new_system()?;
+    fn start_object_server(&mut self) -> Result<blocking::Connection> {
+        let connection = blocking::Connection::system()?;
 
         fdo::DBusProxy::new(&connection)?.request_name(
-            "org.coreos.zincati",
-            fdo::RequestNameFlags::ReplaceExisting.into(),
+            WellKnownName::from_static_str("org.coreos.zincati")?,
+            zbus::fdo::RequestNameFlags::ReplaceExisting.into(),
         )?;
 
-        let mut object_server = zbus::ObjectServer::new(&connection);
-        let experimental_interface = Experimental {
-            agent_addr: self.agent_addr.clone(),
-        };
-        object_server.at(
+        connection.object_server().at(
             &ObjectPath::try_from("/org/coreos/zincati")?,
```

Comment:
You don't need to use that type here, it's kept generic for a reason:

```rust
    "/org/coreos/zincati",
```
would do.